src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedObj: GroupsMap<T> = {};
```

Comment:
`obj` is still the data structure :)
```suggestion
  const groupedItems: GroupsMap<T> = {};
```

Reply:
Thank You !)